addon/components/high-charts.js

```diff
@@ -58,15 +59,16 @@ export default Ember.Component.extend({
   },
 
   draw() {
-    let options = [ get(this, 'buildOptions') ];
-    let mode  = get(this, 'mode');
+    let completeChartOptions = [ get(this, 'buildOptions'), get(this, 'callback') ],
```

Comment:
I think both are fine and my main consideration here is to improve code readability and collaboration. Since you can declare variables anywhere in a function, it's better to keep them at one place so the developers who are not familiar with the code base won't make the trivial mistake to re-declare an existing variable. 

Besides, since var/let support multiple declarations, declaring one let per variable is a waste of code.